### PR TITLE
Add genre director profile views and react router

### PR DESCRIPTION
**Pull request by mdillemuth**
I have a few problems so far with this branch and I believe they are mainly about the 'user' state. 

add-director-view-genre-view

**Things that work**
- Registering a new account
- Removing an account
- Logging in
- Logging out
- Displaying favorites in profile view
- Removing a favorite from profile view
- Displaying movies with same director in director view
- Displaying movies with same genre in genre view

**Things that DONT work**
- The moviecards sometimes do not load immediately after logging in, this is shown in the gif
- Updating account information from profile view
- Refreshing the page while in profile view 
- Immediately updating the user's favorites after removing a favorite while in profile view

### CODE REVIEW

File: src/components/profile-view/profile-view.jsx [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+import { Link } from 'react-router-dom';
+
+const ProfileView = ({ user, movies }) => {
+  const [formData, setFormData] = useState({
+    username: '',
+    email: '',
+    password: '',
+    birthday: '',
+  });
+
```
Comment (eadsimone) [inline src/components/profile-view/profile-view.jsx]: `
componentDidMount() {
		const accessToken = localStorage.getItem('token');
		if (accessToken !== null) {
			this.getUser(accessToken);
		}
	}`

	getUser(token) {
		const username = localStorage.getItem('user');
		axios
			.get(`https://bchanmyflix.herokuapp.com/users/${username}`, {
				headers: { Authorization: `Bearer ${token}` },
			})
			.then((response) => {
				this.setState({
					Username: response.data.Username,
					Password: response.data.Password,
					Email: response.data.Email,
					Birthday: response.data.Birthday,
					FavoriteMovies: response.data.FavoriteMovies,
				});
			})
			.catch(function (error) {
				console.log(error);
			});
	}

File: src/components/main-view/main-view.jsx [hunk 2]
```diff
@@ -1,25 +1,43 @@
...
+  // Authentication and adding 'user' to state
+  handleLoggedIn = (authData) => {
+    this.setState({
+      //  CHANGING THIS FOR EXPERIMENTATION (ORIGINAL: authData.user.Username)
+      user: authData.user,
+    });
+
+    localStorage.setItem('token', authData.token);
+    localStorage.setItem('user', authData.user.Username);
+  };
```
Comment (eadsimone) [inline src/components/main-view/main-view.jsx]: this.getMovies(authData.token);

**Review by eadsimone — CHANGES_REQUESTED** [review]

File: src/components/movie-view/movie-view.jsx [hunk 3]
```diff
@@ -9,45 +11,55 @@ class MovieView extends Component {
...
-              <div className='movie-genre'>
-                <span className='label'>Genre: </span>
-                <span className='value'>{movie.Genre.Name || ''}</span>
+            <div className='col-lg-6 d-flex flex-column align-items-center justify-content-between'>
+              <div className='movie-title d-flex justify-content-center align-items-center w-100 mb-2'>
+                <span className='value h2 text-primary mr-2 font-weight-semi-bold'>
+                  {movie.Title || ''}
+                </span>
+                <p className='h4 text-dark ml-2'>
+                  <i className='star fa fa-star-o text-warning'></i>
```
Comment (eadsimone) [inline src/components/movie-view/movie-view.jsx]: for favorites you can something like:
```
<div className="favorites-button-container">
								<Button className="movie-view-button favorites-button" value={movie._id} onClick={(e) => this.handleAddFavorite(e, movie)}>
									Add to Favorites
								</Button>
							</div>
```

File: src/components/movie-view/movie-view.jsx [hunk 4]
```diff
@@ -9,45 +11,55 @@ class MovieView extends Component {
   }
```
Comment (eadsimone) [inline src/components/movie-view/movie-view.jsx:12]:
```
handleAddFavorite(e, movie) {
		e.preventDefault();
		const username = localStorage.getItem('user');
		const token = localStorage.getItem('token');
		axios({
			method: 'post',
			url: `https://bchanmyflix.herokuapp.com/users/${username}/Movies/${movie._id}`,
			headers: { Authorization: `Bearer ${token}` },
		})
			.then(() => {
				alert(`${movie.Title} was add to Favorites`);
			})
			.catch(function (err) {
				console.log(err);
			});
	}
```

**Comment by mdillemuth** [comment]
3-5-gif-final

**Major Changes**
- Client-side authentication that prevents submitting invalid forms to the server
- Server-side authentication that alerts the user when the received form was invalid (ex. username taken)
- Redirecting from invalid URLs to a '/not-found' page
- Code refactoring and abstracting components to clean-up project
  - Separate components for registration form and login form
  - Separate component for displaying favorites in profileView (favoritesView)
  - Separate component for displaying movies 
-----
**Minor Changes**
  - Added favicon
  - Added customized loading spinner gif for when form is submitted to display while waiting for server response
  - NavBar is abstracted to render on all routes, but profile/logout buttons only render when user after user is logged in

**Review by eadsimone — APPROVED** [review]
urock

**Pull request merged**